### PR TITLE
Define GDNative sizes using sizeof(godot_real_t) and sizeof(int32_t)

### PR DESCRIPTION
**Pull request by aaronfranke**
Previously these values were hard-coded, so it wouldn't work if `sizeof(real_t)` is some other size. I also defined Vector2i/Vector3i/Rect2i with `sizeof(int32_t)` for consistency. I could just add this to #21922 but this can be reviewed and merged by itself so I decided to submit it separately for @vnen and/or @bruvzg to look at.

### CODE REVIEW

File: modules/gdnative/include/gdnative/color.h
Comment (bruvzg) [inline modules/gdnative/include/gdnative/color.h:41]: Why not `sizeof(float) * 4`, for consistency?

File: modules/gdnative/include/gdnative/rect2.h
Comment (bruvzg) [inline modules/gdnative/include/gdnative/rect2.h:50]: `Vector2/3i` members are defined as `int`, which is 32-bit on all supported platforms, but it's "at least
16" according to C++ standard.

**Review by bruvzg — COMMENTED** [review]
`Variant` seems to have some `real_t` dependency as well:
https://github.com/godotengine/godot/blob/d39f6386ce3a7916dbb94fef5ff65e7599e060f0/core/variant/variant.h#L206

File: modules/gdnative/include/gdnative/rect2.h
Comment (aaronfranke) [reply modules/gdnative/include/gdnative/rect2.h:50]: Actually, [Vector3i uses `int32_t`](https://github.com/godotengine/godot/blob/master/core/math/vector3i.h#L46), and I thought Vector2i did the same, but I was mistaken, as [Vector2i uses `int`](https://github.com/godotengine/godot/blob/master/core/math/vector2.h#L268). This inconsistency should probably be fixed.

File: modules/gdnative/include/gdnative/aabb.h
Comment (vnen) [inline modules/gdnative/include/gdnative/aabb.h:38]: These header files under `include` are meant to be used by GDNative libraries, so they should be self-contained and thus cannot depend on `core`. Not sure how to solve this but maybe we could have a `math_defs.h` in GDNative itself to solve what `real_t` is without doing it in every header.

**Comment by vnen** [comment]
Seems okay to me (apart from my comment) but I would prefer this to be done after my rewrite in #44989 is merged.

File: modules/gdnative/include/gdnative/aabb.h
Comment (aaronfranke) [reply modules/gdnative/include/gdnative/aabb.h:38]: Actually, there is already `typedef float godot_real;` in `modules/gdnative/include/gdnative/gdnative.h`. This should probably be changed to this:

```cpp
#ifdef REAL_T_IS_DOUBLE
typedef float godot_real;
#else
typedef float godot_real;
#endif
```

Then we need to move up `#include <gdnative/gdnative.h>` in these headers, and then `sizeof(real_t)` can be `sizeof(godot_real)`.

Comment (vnen) [reply modules/gdnative/include/gdnative/aabb.h]: This was removed. `godot_real` is renamed to `godot_float` and it's always a `double` (since it's meant to be used in Variant which is always 64 bits). We may define a `godot_real_t` to use here instead.

And yes, you cannot include `gdnative.h` before the definition of the structs, that's why I suggested a new header.

> This assumes that people including this would define `REAL_T_IS_DOUBLE` in their GDNative builds if they use doubles.

Yes, people will have to that anyway.

**Review by akien-mga — APPROVED** [review]
Looks good to me, I'll let @vnen do a final review and merge.

**Review by vnen — APPROVED** [review]
LGTM

**Comment by vnen** [comment]
Thanks!

**Pull request merged**
